support/util/networking_test.go

```diff
@@ -0,0 +1,87 @@
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/openshift/hypershift/api/util/ipnet"
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	"k8s.io/utils/pointer"
+)
+
+func TestSetAdvertiseAddress(t *testing.T) {
+	type args struct {
+		hcp      *hyperv1.HostedControlPlane
+		defaults *DefaultAdvIps
+	}
+	tests := []struct {
+		name string
+		args args
```

Comment:
why not include hcp directly here? 
also why do you need to specify defaults for every case?

Reply:
Not following the first comment, fixed the second one.